src/modules/MouseUtils/InclusiveMouse/InclusiveCrosshair.cpp

```diff
@@ -0,0 +1,452 @@
+    HHOOK m_mouseHook = NULL;
+    static LRESULT CALLBACK MouseHookProc(int nCode, WPARAM wParam, LPARAM lParam) noexcept;
+
+    static constexpr auto m_className = L"InclusiveCrosshair";
+    static constexpr auto m_windowTitle = L"PowerToys Inclusive Mouse";
+    HWND m_hwndOwner = NULL;
+    HWND m_hwnd = NULL;
+    HINSTANCE m_hinstance = NULL;
+    static constexpr DWORD WM_SWITCH_ACTIVATION_MODE = WM_APP;
+
+    winrt::DispatcherQueueController m_dispatcherQueueController{ nullptr };
+    winrt::Compositor m_compositor{ nullptr };
+    winrt::Desktop::DesktopWindowTarget m_target{ nullptr };
+    winrt::ContainerVisual m_root{ nullptr };
+    winrt::LayerVisual m_crosshair_border_layer{ nullptr };
+    winrt::LayerVisual m_crosshair_layer{ nullptr };
+    winrt::SpriteVisual m_left_crosshair_border{ nullptr };
+    winrt::SpriteVisual m_left_crosshair{ nullptr };
+    winrt::SpriteVisual m_right_crosshair_border{ nullptr };
+    winrt::SpriteVisual m_right_crosshair{ nullptr };
+    winrt::SpriteVisual m_top_crosshair_border{ nullptr };
+    winrt::SpriteVisual m_top_crosshair{ nullptr };
+    winrt::SpriteVisual m_bottom_crosshair_border{ nullptr };
+    winrt::SpriteVisual m_bottom_crosshair{ nullptr };
+
+    bool m_visible = false;
+    bool m_destroyed = false;
+
+    // Configurable Settings
+    winrt::Windows::UI::Color m_crosshair_border_color = INCLUSIVE_MOUSE_DEFAULT_CROSSHAIR_BORDER_COLOR;
+    winrt::Windows::UI::Color m_crosshair_color = INCLUSIVE_MOUSE_DEFAULT_CROSSHAIR_COLOR;
+    float m_crosshair_radius = INCLUSIVE_MOUSE_DEFAULT_CROSSHAIR_RADIUS;
+    float m_crosshair_thickness = INCLUSIVE_MOUSE_DEFAULT_CROSSHAIR_THICKNESS;
+    float m_crosshair_border_size = INCLUSIVE_MOUSE_DEFAULT_CROSSHAIR_BORDER_SIZE;
+    float m_crosshair_opacity = max(0.f, min(1.f, (float)INCLUSIVE_MOUSE_DEFAULT_CROSSHAIR_OPACITY / 100.0f));
+};
+
+InclusiveCrosshair* InclusiveCrosshair::instance = nullptr;
+
+bool InclusiveCrosshair::CreateInclusiveCrosshair()
+{
+    try
+    {
+        // We need a dispatcher queue.
+        DispatcherQueueOptions options = {
+            sizeof(options),
+            DQTYPE_THREAD_CURRENT,
+            DQTAT_COM_ASTA,
+        };
+        ABI::IDispatcherQueueController* controller;
+        winrt::check_hresult(CreateDispatcherQueueController(options, &controller));
+        *winrt::put_abi(m_dispatcherQueueController) = controller;
+
+        // Create the compositor for our window.
+        m_compositor = winrt::Compositor();
+        ABI::IDesktopWindowTarget* target;
+        winrt::check_hresult(m_compositor.as<ABI::ICompositorDesktopInterop>()->CreateDesktopWindowTarget(m_hwnd, false, &target));
+        *winrt::put_abi(m_target) = target;
+
+        // Our composition tree:
+        //
+        // [root] ContainerVisual
+        // \ [crosshair border layer] LayerVisual
+        //   \ [crosshair border sprites]
+        //     [crosshair border layer] LayerVisual
```

Comment:
Is it correct that it's the same as two lines before?

Reply:
Nice spotting, should've been "crosshair layer".